modyn/config/schema/pipeline/trigger/cost/cost.py

```diff
@@ -0,0 +1,74 @@
+"""Cost aware triggers are evaluate the trade-off between the cost of a trigger
+(e.g. measured by training machine time) and the benefit of the trigger (e.g.
+measured by a regret metric like data incorporation latency or avoidable
+misclassification incorporation latency)."""
+
+from typing import Literal
+
+from pydantic import Field
+
+from modyn.config.schema.base_model import ModynBaseModel
+
+from ..performance.performance import _InternalPerformanceTriggerConfig
+
+
+class _CostTriggerConfig(ModynBaseModel):
+    """Base class for cost aware trigger policies."""
+
+    cost_tracking_window_size: int = Field(
+        description="How many trigger into the past should be considered for the linear cost w.r.t. data amount model."
+    )
+
+
+class DataIncorporationLatencyCostTriggerConfig(_CostTriggerConfig):
+    """Cost aware trigger policy configuration that uses the data incorporation
+    latency as a regret metric.
+
+    While no trigger occurs the un-integrated data is cumulated and the sum of all seconds of every unincorporated data
+    point is added up.
+
+    `incorporation_delay_per_training_second` servers as a conversion rate between budget (training time) and regret
```

Comment:
I don't understand this, unfortunately. Is this our threshold that our cumulative metriic may not cross? I don't understand yet how you can convert between training time and incorporation latency.

Reply:
From https://cse.hkust.edu.hk/~weiwa/papers/continuum-socc18.pdf (p. 7):

> we propose a cost-aware policy for fast data incorporation at low training cost. We model the cost sensitivity of a user with a _**“knob” parameter w**_

`incorporation_delay_per_training_second` reflects this "knob parameter w".

Reply:
The way I understand it, `DataIncorporationLatencyCostTriggerConfig` triggers when the accumulated regret (sum over all latencies as I mentioned in another comment) crosses a threshold. But I start to believe I am wrong in this assumption. I don't understand yet what the role of this w in the Modyn setting is. Because we say the training cost is linearily dependend on the number of samples, I still don't get the role training time plays here, and why we need to convert between training time and this latency.